test/reservation_test.rb

```diff
@@ -0,0 +1,103 @@
+require_relative 'test_helper'
+
+describe "Reservation class" do
+  before do
+    @date = Date.parse("2020-02-01")
+    @room = Hotel::Room.new(10,200)
+    @reservation = Hotel::Reservation.new(1, @date, 10, "confirmed", @room)
+
+    date2 = Date.parse("2020-02-15")
+    room = Hotel::Room.new(10,200)
+    @reservation2 = Hotel::Reservation.new(1, date2, 10, "confirmed", @room)
+
+    date3 = Date.parse("2020-02-05")
+    room = Hotel::Room.new(10,200)
+    @reservation3 = Hotel::Reservation.new(1, date3, 10, "confirmed", @room)
+
+    @confirmed_reservations = [@reservation, @reservation2, @reservation3]
+    
+    temp_date1 = Date.parse("2020-01-29")
+    @reservation_request1 = Hotel::Reservation.new(1, temp_date1, 3)
+
+    temp_date2 = Date.parse("2020-01-31")
+    @reservation_request2 = Hotel::Reservation.new(1, temp_date2, 3)
+
+    temp_date3 = Date.parse("2020-02-02")
+    @reservation_request3 = Hotel::Reservation.new(1, temp_date3, 3)
+
+    temp_date4 = Date.parse("2020-02-09")
+    @reservation_request4 = Hotel::Reservation.new(1, temp_date4, 3)
+
+    temp_date5 = Date.parse("2020-02-11")
+    @reservation_request5 = Hotel::Reservation.new(1, temp_date5, 3)
+  end
+
+  describe "Reservation Instantiation" do
+    it "is an instance of Reservation" do
+      expect(@reservation).must_be_kind_of Hotel::Reservation
+    end
+
+    it "validates check in date" do
+      expect{Hotel::Reservation.new(1, "date", 10, "confirmed", @room)}.must_raise ArgumentError
+    end
+
+    it "validates number of days" do
+      expect{Hotel::Reservation.new(1, @date, -1, "confirmed", @room)}.must_raise ArgumentError
+    end
+
+    it "assigns room" do
+      expect(@reservation.room.room_number).must_equal 10
+    end
+
+    it "assigns room number to the room" do
+      expect(@reservation.room_number).must_equal 10
+    end
+
+    it "creates a reservation request" do
+      expect(@reservation_request1.status).must_equal :pending
+      assert_nil(@reservation_request1.room)
+      assert_nil(@reservation_request1.room_number)
```

Comment:
You can use `must_be_nil` here:

```suggestion
      expect(@reservation_request1.room).must_be_nil
      expect(@reservation_request1.room_number).must_be_nil
```